libtransmission/torrent.cc

```diff
@@ -1773,10 +1773,16 @@ void tr_torrent::recheck_completeness()
         }
 
         completeness_ = new_completeness;
-        session->close_torrent_files(id());
+
+        if (is_done() && was_running && session->shouldFullyVerifyCompleteTorrents())
```

Comment:
I'm not 100% sure there aren't any cases where this condition will be true when we don't want it to be.

I think we can let this through, but we should keep an eye out.

Reply:
I've been using that feature for more than a year (it's included on my main branch of https://github.com/Coeur/swiftmission). And it does what it's advertising: it triggers verify data once on download completion. If there were corruption, it will continue the download and re-trigger verify data on second download completion.

Reply:
Cool.